community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/InternalAutoIndexOperations.java

```diff
@@ -0,0 +1,316 @@
+                            throws LegacyIndexNotFoundKernelException, EntityNotFoundException
+                    {
+                        ops.relationshipAddToLegacyIndex( InternalAutoIndexing.RELATIONSHIP_AUTO_INDEX, entityId,
+                                keyName, value );
+                    }
+
+                    @Override
+                    public void remove( DataWriteOperations ops, long entityId, String keyName, Object value )
+                            throws LegacyIndexNotFoundKernelException, EntityNotFoundException
+                    {
+                        ops.relationshipRemoveFromLegacyIndex( InternalAutoIndexing.RELATIONSHIP_AUTO_INDEX,
+                                entityId, keyName, value );
+                    }
+
+                    @Override
+                    public void remove( DataWriteOperations ops, long entityId, String keyName )
+                            throws LegacyIndexNotFoundKernelException, EntityNotFoundException
+                    {
+                        ops.relationshipRemoveFromLegacyIndex( InternalAutoIndexing.RELATIONSHIP_AUTO_INDEX,
+                                entityId, keyName );
+                    }
+
+                    @Override
+                    public void remove( DataWriteOperations ops, long entityId )
+                            throws LegacyIndexNotFoundKernelException, EntityNotFoundException
+                    {
+                        ops.relationshipRemoveFromLegacyIndex( InternalAutoIndexing.RELATIONSHIP_AUTO_INDEX, entityId );
+                    }
+
+                    @Override
+                    public void ensureIndexExists( DataWriteOperations ops )
+                            throws LegacyIndexNotFoundKernelException, EntityNotFoundException
+                    {
+                        ops.relationshipLegacyIndexCreateLazily( InternalAutoIndexing.RELATIONSHIP_AUTO_INDEX, null );
+                    }
+                };
+
+        public abstract void add( DataWriteOperations ops, long entityId, String keyName, Object value )
+                throws LegacyIndexNotFoundKernelException, EntityNotFoundException;
+
+        public abstract void remove( DataWriteOperations ops, long entityId, String keyName, Object value )
+                throws LegacyIndexNotFoundKernelException, EntityNotFoundException;
+
+        public abstract void remove( DataWriteOperations ops, long entityId, String keyName )
+                throws LegacyIndexNotFoundKernelException, EntityNotFoundException;
+
+        public abstract void remove( DataWriteOperations ops, long entityId )
+                throws LegacyIndexNotFoundKernelException, EntityNotFoundException;
+
+        public abstract void ensureIndexExists( DataWriteOperations ops )
+                throws LegacyIndexNotFoundKernelException, EntityNotFoundException;
+    }
+
+    private AtomicReference<Set<String>> propertyKeysToInclude = new AtomicReference<>( Collections.emptySet() );
```

Comment:
if we will add propertyKeysToInclude to constructor we should be able to make it just a plain set